### ISSUE TITLE
Rule proposal: one-export

### ISSUE THREAD

**Issue by g-plane** [eslint/eslint#10380]
**Please describe what the rule should do:**
Enforce exporting something once.

It's possible to export some public interfaces many times both in ES6 modules and CommonJS. Like this:

```js
// ES6 modules

export const foo = 'foo'
// Other code here...
export function bar() {}
```

```js
// CommonJS

module.exports.foo = 'foo'
// Other code here...
module.exports.bar = function bar() {}
```

However, it's hard to catch all the available public interfaces of a file, especially when the file is so large and many interfaces are exported.

The better way for readability is like this:

```js
// ES6 modules

const foo = 'foo'
// Other code here...
function bar() {}
export { foo, bar }
```

```js
// CommonJS

const foo = 'foo'
// Other code here...
function bar() {}
module.exports = { foo, bar }
```

Some questions:
1. If both ES6 modules and CommonJS are used, should it be treated as exporting something twice? (My personal opinion: This should be treated as exporting twice.)
Example:
```js
const foo = 'foo'
function bar() {}
export { foo }
module.exports = { bar }
```

2. Is this rule possibly fixable?

**What category of rule is this? (place an "X" next to just one item)**

[ ] Enforces code style
[ ] Warns about a potential error
[x] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

Described above.

**Why should this rule be included in ESLint (instead of a plugin)?**
Seems no plugins are working on this.
